### ISSUE TITLE
Why does GPUs use more memory than "memory" in log and each GPU uses different memory in Distributed training?

### ISSUE THREAD

**Issue by yy9996** [open-mmlab/mmdetection3d#1573]
As picture showed.

![image](https://user-images.githubusercontent.com/102511364/174748550-703aa744-4eb3-4a80-b6e1-f31bb93557e9.png)

**Comment by Tai-Wang** [comment]
The memory shown in the log is computed by pytorch while the memory observed from nvidia-smi is computed by nvidia. Different memories used across GPUs can be caused by many factors, e.g., different data augmentations resulting in different input sizes.

**Comment by yy9996** [comment]
Thanks! And I want to ask If the memory keeps going up, does it mean there is redundant data or a memory leak?

> The memory shown in the log is computed by pytorch while the memory observed from nvidia-smi is computed by nvidia. Different memories used across GPUs can be caused by many factors, e.g., different data augmentations resulting in different input sizes.

**Comment by Tai-Wang** [comment]
Yes, if the memory keeps going up, it is likely that there is something wrong with the procedure of computing gradients for backward propagation or somewhere else.

**Comment by yy9996** [comment]
thanks a lot

**Comment by JZhong-Li** [comment]
I also encountered this problem, and it seems to be more obvious. As shown in below. Is there something inappropriate in my code, or what I should do to solve this issue. Thanks.
![mmexport1677422487775](https://user-images.githubusercontent.com/76928254/221417650-bf9d26f3-128c-4230-bdf9-5ef2d1f0fbb8.jpg)
![mmexport1677422484528](https://user-images.githubusercontent.com/76928254/221417666-18428d45-3da2-4d49-8ff7-38169a3ee6a1.png)
